cmd/ndm_daemonset/probe/mountprobe.go

```diff
@@ -106,3 +128,36 @@ func (mp *mountProbe) FillBlockDeviceDetails(blockDevice *blockdevice.BlockDevic
 		blockDevice.FSInfo.FileSystem = basicMountInfo.FileSystem
 	}
 }
+
+func (mp *mountProbe) setupEpoll(events ...epoll.EventType) error {
+	ep, err := epoll.New()
+	if err != nil {
+		return err
+	}
+	mp.epoll = &ep
+	if len(events) == 0 {
+		events = append(events, epoll.EPOLLERR, epoll.EPOLLPRI)
+	}
+	return ep.AddWatcher(epoll.Watcher{
+		FileName:   mp.mountsFileName,
+		EventTypes: events,
+	})
+}
+
+func (mp *mountProbe) listen() {
+	eventCh, err := mp.epoll.Start()
+	if err != nil {
+		klog.Errorf("error while starting epoll: %v", err)
+		return
+	}
+	defer mp.epoll.Stop()
+	klog.Info("started mount change detection loop")
+	msg := controller.EventMessage{
+		Action:  string(MountEA),
+		Devices: []*blockdevice.BlockDevice{&blockdevice.AllBlockDevices},
+	}
+
+	for range eventCh {
```

Comment:
What is the purpose of ranging over the channel here? 

This causes to read from the channel which is not needed.

Reply:
nvm. Understood the logic after going through the epoll package. A message to scan all devices will be sent on the udev channel whenever a change is detected in mounts file (this change is detected and send over eventCh)? right?

Reply:
yes.